pluginLoader.js

```diff
@@ -33,7 +33,12 @@ define(function(require, exports, module){
             var _lint = plugin.lint;
 
             function lint(text, settings) {
-                return Promise.resolve(_lint(text, settings));
+                var results = _lint(text, settings,pluginsMeta);
+                if(results.promise !== undefined && results.promise !== null){
```

Comment:
I couldn't find where result gets a promise added as a property...  Why is that needed?